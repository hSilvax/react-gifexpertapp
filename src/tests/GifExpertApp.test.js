const { shallow } = require("enzyme");
const { default: GifExpertApp } = require("../GifExpertApp");

describe("Pruebas de <GifExpertApp/>", () => {
    
    test("Debe de comparar con el snapshot", () => {
        
        const wrapper = shallow(<GifExpertApp/>);

        expect(wrapper).toMatchSnapshot();

    });

    test("Debe de mostrar una lista de categoria", () => {
        
        const categories = ["Macross","Rio"];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("GifGrid").length).toBe(categories.length);

    });
    
    

})
