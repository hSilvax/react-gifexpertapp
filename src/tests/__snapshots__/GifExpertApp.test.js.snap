// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <GifExpertApp/> Debe de comparar con el snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;

exports[`Pruebas de <GifExpertApp/> Debe de mostrar una lista de categoria 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="Macross"
      key="Macross"
    />
    <GifGrid
      category="Rio"
      key="Rio"
    />
  </ol>
</Fragment>
`;
